libp2p/host/ping.py

```diff
@@ -45,16 +66,65 @@
     return True
 
 
-async def handle_ping(stream: INetStream) -> None:
-    """``handle_ping`` responds to incoming ping requests until one side errors
-    or closes the ``stream``."""
-    peer_id = stream.muxed_conn.peer_id
+class PingService:
+    """PingService executes pings and returns RTT in miliseconds."""
 
-    while True:
+    def __init__(self, host: IHost):
+        self._host = host
+
+    async def ping(self, peer_id: PeerID) -> int:
+        stream = await self._host.new_stream(peer_id, (ID,))
         try:
-            should_continue = await _handle_ping(stream, peer_id)
-            if not should_continue:
-                return
+            rtt = await _ping(stream)
+            await _close_stream(stream)
+            return rtt
         except Exception:
-            await stream.reset()
-            return
+            await _close_stream(stream)
+            raise
+
+    async def ping_loop(
+        self, peer_id: PeerID, ping_amount: Union[int, float] = math.inf
+    ) -> "PingIterator":
+        stream = await self._host.new_stream(peer_id, (ID,))
+        ping_iterator = PingIterator(stream, ping_amount)
+        return ping_iterator
+
+
+class PingIterator:
+    def __init__(self, stream: INetStream, ping_amount: Union[int, float]):
+        self._stream = stream
+        self._ping_limit = ping_amount
+        self._ping_counter = 0
+
+    def __aiter__(self) -> "PingIterator":
+        return self
+
+    async def __anext__(self) -> int:
+        if self._ping_counter > self._ping_limit:
+            await _close_stream(self._stream)
+            raise StopAsyncIteration
+
+        self._ping_counter += 1
+        try:
+            return await _ping(self._stream)
+        except trio.EndOfChannel:
+            await _close_stream(self._stream)
+            raise StopAsyncIteration
+
+
+async def _ping(stream: INetStream) -> int:
+    ping_bytes = secrets.token_bytes(PING_LENGTH)
+    before = int(time.time() * 10 ** 6)  # convert float of seconds to int miliseconds
```

Comment:
what do you think about just keeping the native `time.time()` value as a `float` for as long as possible (possibly just letting the consumer of the rtt convert to `int` if _they_ want...)

Reply:
First of all thank you for your comments. I learn a lot from them!

I think it is a good idea to keep it as float. Do you think it is better to leave it also as native seconds? I think I prefer to return it as a miliseconds float